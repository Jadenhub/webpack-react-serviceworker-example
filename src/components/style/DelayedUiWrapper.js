import React from 'react';
import styled, { keyframes } from "styled-components";

const makeVisible = keyframes`
  to {
    visibility: visible;
  }
`;

const Wrapper = styled.div`
  animation: ${props => `0s linear ${props.delay}s forwards`} ${makeVisible};
  visibility: hidden;
  display: ${(props) => props.inline ? 'inline-block' : 'block'};
`

const DelayedUiWrapper = ({ children, delay = 0.5, inline = false }) => (
  <Wrapper delay={delay} inline={inline}>{children}</Wrapper>
)

export default DelayedUiWrapper
