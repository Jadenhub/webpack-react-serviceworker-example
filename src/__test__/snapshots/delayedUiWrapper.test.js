import renderer from 'react-test-renderer';
import DelayedUiWrapper from '../../components/style/DelayedUiWrapper';
import React from 'react';

test('renders correctly', () => {
  const tree = renderer
    .create(<DelayedUiWrapper/>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

test('renders correctly', () => {
  const tree = renderer
    .create(<DelayedUiWrapper inline={true}/>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});