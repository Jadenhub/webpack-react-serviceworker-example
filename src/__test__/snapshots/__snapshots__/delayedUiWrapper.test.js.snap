// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="DelayedUiWrapper__Wrapper-qcmkwu-0 iGMkoW"
/>
`;

exports[`renders correctly 2`] = `
<div
  className="DelayedUiWrapper__Wrapper-qcmkwu-0 gvNTfy"
/>
`;
